x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticQueryBuilder.java

```diff
@@ -75,8 +75,13 @@ public class SemanticQueryBuilder extends AbstractQueryBuilder<SemanticQueryBuil
     private final SetOnce<InferenceServiceResults> inferenceResultsSupplier;
     private final InferenceResults inferenceResults;
     private final boolean noInferenceResults;
+    private final boolean throwOnUnsupportedField;
 
     public SemanticQueryBuilder(String fieldName, String query) {
+        this(fieldName, query, true);
+    }
+
+    public SemanticQueryBuilder(String fieldName, String query, boolean throwOnUnsupportedField) {
```

Comment:
Why do we need `throwOnUnsupportedField`? I thought the filter clause in the boolean query ensures that the semantic query is only generated for indices with a semantic text field?

Reply:
Nope. Without this, we'll get an error. 

You can also see this by manually running the query: 
```
GET my-index-*/_search
{
  "query": {
    "bool": {
      "should": [
        {
          "bool": {
            "filter": [
              {
                "terms": {
                  "_index": [
                    "my-index-1"
                  ]
                }
              }
            ],
            "should": {
              "semantic": {
                "field": "inference_field",
                "query": "test"
              }
            }
          }
        }
      ]
    }
  }
}
```

Reply:
So then the `filter` clause must be applied as a post-filter. I think this has potential performance implications, as it means that we're executing this `semantic` query on all indices, potentially multiple times if we're generating multiple semantic query clauses.

One way to fix this would be to add an optional param to the `semantic` query builder indicating the indices to filter on. Then we could use this to pre-filter on index name during data node rewrite.

Reply:
Do you think that's an optimization we could add in a followup ticket?

Reply:
I think that works, but it would be nice to have that done in time for 8.18

Reply:
Please see my comment on https://github.com/elastic/elasticsearch/pull/117839#discussion_r1873717534
Adding a lenient parameter here is totally fine.

Reply:
Thanks, will go that route and rename to `lenient`

Reply:
Addressed in b09e1d0a4ab98beb217ba3a7ff5d3270b1012231